src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,119 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      double xTolerance) {
```

Comment:
Since we gave the first and second parameter a semantic meaning (being the lower and upper bound) we should ensure that the first is actually lower than or equal to the second one.
```
verify(lowerBound <= upperBound, "brent: The lower bound must be less than or equal to the upper bound.")
```